### DISCUSSION TITLE
프롤로그 서비스 / 학습 관련 개선하거나 개발하고 싶은 아이템

### DISCUSSION THREAD

**Discussion by seokhongkim** [woowacourse/prolog#1231] [Ideas]
페어별로 정리해서 코멘트 남겨주세요.

**Comment by Creative-Lee** [comment]
## FE_ 도담 + 도밥 개선사항

반응형 디자인에 집중해서 개선사항 찾아봤습니다~😀

이미지가 포함되어 있어, 노션 링크로 대체하겠습니다.

[반응형 디자인 개선사항](https://www.notion.so/creativedobob/FE_-04-14-c78408ea8f0f463d948d8a7d7fde566d?pvs=4)

**Comment by GC-Park** [comment]
## FE_해온 + 패트릭 개선사항

- 학습로그 검색어에 닉네임을 치면 나오도록 수정

- 프롤로그 프론트엔드 로드맵을 추가

- 프롤로그 안드로이드 로드맵을 추가

- 반응형 시 글자 잘리는 에러 수정

- 프롤로그 백엔드 로드맵 상위 키워드 선택 글자 한줄로 표시되게 수정

- 스크롤을 조금 더 이쁘게 수정

- 글쓰기에 사진을 좀 더 편하게 넣도록 수정

- 학습로그 멤버 검색 창에서 이름 검색하면 바로 적용되도록 수정

- header에 글쓰기 버튼 클릭 후 nav bar 마우스 커서 안되는 이슈 해결

- 스크랩 창에서 ‘스크랩한 글이 없습니다’ 위치 변경

- 학습로그 sessions와 missions 둘 다 내림차순으로 변경

- 최신 학습로그와 인기 있는 학습로그 사이 margin 추가

- 로드맵 밑에 margin 추가

**Reply by hae-on** [reply]
- header에 글쓰기 버튼 클릭 후 nav bar 마우스 커서 안되는 이슈 해결

<img src="https://user-images.githubusercontent.com/80464961/232402550-e4767764-1859-4f31-b8de-1041067c4574.jpg" />

한 번 클릭해야 cursor가 pointer로 바뀝니다. 
<img src="https://user-images.githubusercontent.com/80464961/232402562-9390f6fc-701f-40c0-863e-11c98ff3a344.jpg" />

**Comment by n0eyes** [comment]
### FE_노아이즈 + 참새 개선사항


- 검색 기능 개선
  -  Shadow를 검색 할 경우 Shadow가 포함된 모든 게시글이 노출되지 않는다
  - 작성자 이름으로 검색이 불가능하다

- 토큰 시간이 너무 짧다
- 작성 중 글 유지
- 댓글, 좋아요 알림 표시
- 이미지 첨부 기능
- 인기 학습로그 알고리즘..이 궁금해요..? (계속 보이던 글만 보이는 것 같아요)
- 인기 학습로그 스켈레톤
- 반응형 개선
- 공지사항 목록 추가
- 자기소개 초기화가 불가능하다

**Comment by feb-dain** [comment]
## 🍗FE_야미 | 🧑🏻‍💻FE_솔로스타 개선사항

1. 로그인이 자주 풀리는 문제 개선 (1시간이 너무 짧다는 의견, 글을 작성하다가 로그인이 풀리면 글이 날아가는 문제)
2. 검색 기능 개선 - 정확한 키워드 검색이 되지 않는다. (ex. 자기소개 검색x, 검색 결과가 연관성이 너무 떨어짐)
3. 모바일 미디어 쿼리 개선 (글자가 세로 정렬됨)
4. 셀렉트 박스 순서 변경 (최신순으로) 
5. 임시저장 기능 개선 (임시저장이 잘 되지 않음)
6. 로드맵 > 프론트엔드 상위/하위 키워드가 없는 문제 개선
7. 로드맵 상위 키워드 CSS 개선 
8. 로드맵 키워드 순서 개선
9. 인기있는 학습로그 카드 안 가로 스크롤 제거
10. 학습로그 이름 영역 CSS 개선
11. 스크롤 디자인 개선
12. 에러 알림이 제대로 표시가 안 되는 문제 개선 (ex. 제목 글자 수 50자 이하 알림이 떠야하는데 오류 발생했다는 알림이 뜸)
13. 자주 사용되는 태그를 최상단에 보여주도록 개선

<br>

### 🪄 건의 사항
1. 댓글 알림 기능 추가
2. 다크모드 추가
3. 캐러셀 일시정지 좀 더 키우기
4. TOC 기능 추가 
5. 프로필에 하트한 게시글 모아볼 수 있는 기능 추가
6. 추천 게시글 기능 추가
7. 리드미와 소개글 연동 기능 추가
8. 관심있는 태그를 설정하고, 관심있는 태그를 설정한 게시글이 업로드되면 알람 받는 기능 추가
9. 학습로그 이름 영역을 클릭하면 해당 멤버가 작성한 게시글을 볼 수 있는 기능 추가

**Reply by kang-hyungu** [reply]
리드미(깃헙)와 소개글(프롤로그) 연동 기능 추가

**Reply by feb-dain** [reply]
* 셀렉트 박스 순서 (현재 오래된 순인데 최신순으로 변경)
* 건의사항 6. 만약 어떤 사람의 커스텀 훅 관련 글을 읽고 있으면 다른 사람이 작성한 커스텀 훅 글들도 추천해주는 기능

**Comment by mcodnjs** [comment]
## BE_라온, 오션, 아마란스 프롤로그 개선안
### 🔴P1(우선순위 높음)
1. **세션 시간 연장**
    - 최소 1시간 이상
2. **검색 기능 개선**
    - 5기들의 글이 검색이 되지 않음
    - 우선순위: (시간순), 제목 → 본문
3. **글 이미지 첨부**
4. **글 작성 기능**
    - 밑줄, URI 치환, 코드블럭
5. **반응형 UI**
    - 글자 줄바꿈 현상이 일어남
    - 모바일 크기에서는 배치를 바꾸거나 폰트 크기를 줄이는 걸로
6. **필터 기능**
    - 최신순 정렬
    - 태그 컨벤션 통일 (대소문자, 한글, #)
7. **공지 기능 및 학습로그 소개**
    - 업데이트 요소 작성
    - 공지사항 작성
8. **임시저장**
    - 해당 기능의 접근성이 떨어짐
        - 현재: 글쓰기 후 임시저장을 누르고, 다시 글쓰기 버튼을 클릭하면 임시저장 글쓰기 가능
        - 제안1: 임시저장 학습로그를 프로필에서 볼 수 있도록 (적은 학습로그를 현재 프로필에서 볼 수 있는 것처럼)
        - 제안2: 글쓰기 버튼을 눌렀을 때, 과거의 임시저장한 글들의 목록을 볼 수 있게
    - 1개의 글만 임시저장 된다는 단점
9. **인기 있는 학습로그**
    - 업데이트 되는 주기 및 기준 개선
    - 어떤 이유인진 모르겠으나 몇몇 기기에서 학습로그 카드에 사진처럼 스크롤 바가 생김
![image](https://user-images.githubusercontent.com/64852591/231995043-9dbdf96c-c7a7-4953-b3a8-9e038b91d548.png)
10. **로드맵 디자인**
    - 뷰 크기가 작아서 글씨가 개행 되는 문제
    - 현재 로드맵
       - 형식: 커리큘럼 → 세션 → 상위 키워드 → 하위 키워드 → 하하위 키워드 → Quiz
       - 고려사항: 상위/하위/하하위 키워드의 개수가 레벨마다 명확하지 않음 → 고려사항에 영향을 안받기 위해서는 세로 형식 유지 필요
    - 제안1. 상위 키워드에서 하위 키워드를 접었다가 펼 수 있게 하기(노션의 들여쓰기 구조처럼)
        ```java
        상위 키워드
            ㄴ하위 키워드1
            ㄴ하위키워드2
        ```
    - 제안2. 마인드맵 형식의 UI

### 🟡P2(우선순위 중간)

1. **로드맵 기능**
    - 학습로그와 같은 댓글 , 스크랩, 좋아요 기능
    - 로드맵 답변 글의 미리보기 텍스트에도 마크다운이 적용되었으면 좋겠다. ⇒ 마크다운 적용이 힘들다면 특수기호 제거라도
![image](https://user-images.githubusercontent.com/64852591/232015133-83defb86-4c66-4a07-b86c-152c328ce24c.png)    

### 🔵P3(우선순위 낮음 - 희망사항)

1. **다크모드 or 컬러테마**
    - 사용자에게 커스텀 환경 제공
2. **스크롤바 예쁘게 디자인**
3. **프롤로그 조회수**
    - 새로고침 시 계속해서 증가하는 문제가 있음
    - 스크랩 및 좋아요 설정,취소 시 조회수가 증가함
    - 로드맵 조회수 없음
4. **폰트**
    - 제목이나 태그같은 요소에는 다른 폰트를 적용
5. **홈 배너 잘림**
    - 맨 우측 끝에 이후 배너가 나옴 (환경마다 다른 듯)
6. **무한스크롤**
    - 기존의 페이지네이션 방식을 무한스크롤 방식으로 변경하는건?
7. **댓글 알림 기능**
    - 본인의 학습 로그에 달린 댓글에 대한 알림
    - 대댓글 기능도 있으면 👍
8. **검색**
    - 상단바에 검색 기능 추가
9. **태그 사이 띄어쓰기**
![image](https://user-images.githubusercontent.com/64852591/232015217-c1a4de86-9e25-4306-9bc0-b3df68bd9805.png)
10. **크루들의 코스/기수 정보 노출**
    - ex) 아마란스 - 5기 백엔드
---
### 우테코 학습 고도화 관련 제안
- 레벨 로그를 올리는 곳을 통일하는 것은 어떤지
    - pdf, 블로그 링크, PROLOG 학습로그 등 크루 마다 레벨로그를 업로드하는 방식이 천차만별임
    - PROLOG 학습로그로 올리도록 통일하는 것은 어떤지
    - 만약 통과된다면 PROLOG에 레벨로그를 올리는 카테고리가 따로 있었으면 좋겠다!
        (레벨로그 올라오는 시기엔 다른 학습로그들이 많이 묻혀서…)

**Comment by Cl8D** [comment]
# BE_푸우, 호이, 져니 프롤로그 개선안

---

## ✔️ 검색 조건 필터링

### **현재의 문제점**
<img width="1107" alt="이미지1" src="https://user-images.githubusercontent.com/46810837/232383494-c871ab11-1d60-459e-a86c-5eb514f849cf.png">
현재는 단순히 검색어로 검색을 진행하고, 결과에 대해 최신순으로 정렬이 되고 있어서 원하는 데이터를 더 다양하게 볼 수 없다는 점이 불편했습니다.

### **개선 사항**
최신순, 댓글이 많은 순, 조회순, 하트순, 스크랩순
이렇게 4가지 기능을 추가해서 세부 정렬 조건을 두면 좋을 것 같습니다.

---

## ✔️ 대댓글 기능 추가

### 현재의 문제점
<img width="708" alt="스크린샷 2023-04-16 오후 10 51 37" src="https://user-images.githubusercontent.com/46810837/232383555-a2de5935-2127-42e7-b952-5940ced70bfb.png">
현재는 단순히 댓글만 달 수 있고, 달린 댓글에 대해서 답글을 달 수 있는 기능이 없습니다.

### 개선 사항
선택한 댓글에 대해서 답글을 달 수 있는 기능이 추가되면 좋을 것 같습니다.
또한, 댓글의 depth를 0으로 두었을 때, 해당 댓글의 답글을 depth = 1이라고 둔다면, 해당 답글 밑에 달리는 댓글들은 모두 depth=1을 가지도록 하면 좋을 것 같습니다.

EX. 티스토리 댓글 기능
<img width="834" alt="스크린샷 2023-04-16 오후 10 53 46" src="https://user-images.githubusercontent.com/46810837/232383596-f60ebdef-8101-4501-b5d7-78cf5e99fdc2.png">
또한, 이렇게 된다면 원본 댓글을 제거하였을 때 하단 댓글을 모두 제거할 것인지, 혹은 해당 댓글의 content만 ‘삭제된 댓글입니다.’라고 뜨고 하단의 답글은 남겨둘 것인지도 체크해야 할 것 같습니다.
개인적인 의견으로는 후자가 더 좋을 것 같습니다. (하위 댓글의 작성자가 다름에도 다른 유저에 의해 본인의 댓글이 지워지도록 하는 것은 좋지 않다고 생각해서)

---

## ✔️ 내 정보

### 현재의 문제점
<img width="239" alt="스크린샷 2023-04-16 오후 10 55 49" src="https://user-images.githubusercontent.com/46810837/232383640-a7429c4d-945d-4936-add4-a9966e59843b.png">
현재는 닉네임에 대해서만 수정이 가능하고, 프로필 이미지에 대해서는 수정이 불가능합니다. 또한, 자신에 대한 정보가 너무 부족하다는 생각이 들었습니다…!

### 개선 사항

아마 깃허브의 프로필 이미지를 바로 긁어오면서 로드해주는 형식인 것 같은데, 프로필 이미지에 대해서도 자체 DB에서 관리하도록 해서 (ex. S3 url) 이미지에 대한 수정도 이루어질 수 있도록 했으면 좋겠습니다.

추가적으로 생각난 거지만, 개인정보에서 CREW라고만 나오지 않고, 기수와 과정 정보도 함께 분류하여 출력해주면 좋을 것 같습니다.

다만 이 문제를 해결하기 위해서는 기존 데이터에 대한 마이그레이션은 어떻게 해야 할지가 고민인 부분 같은데… 깃허브 아이디로 가입을 하는 방식이기 때문에 깃허브 정보를 매칭시키는 식으로 마이그레이션을 할 수 있지 않을까 싶습니다.

그리고 신규 사용자에 대해서는 별도로 입력 or 내부적으로 데이터 매칭…? 하는 형식으로 해도 좋을 것 같습니다.

---

## ✔️ 학습로그 검색 - Member 검색

### 현재의 문제점
<img width="632" alt="스크린샷 2023-04-16 오후 10 59 11" src="https://user-images.githubusercontent.com/46810837/232383748-609a8319-155f-45ba-82ad-45a1660fb2f2.png">

현재 member에 대해 검색을 할 때 모든 멤버들을 기준으로 검색이 됩니다.
또한, 출력되는 크루 리스트의 조건이 불명확합니다. (아이디는 아니고…. 사전순도 아니고…?)

### 개선 사항

이전 기수와 동일한 이름을 가진 크루가 존재하기도 해서, 기수별로 1차 필터링을 하는 것도 좋을 것 같습니다. 
또한, 하단 리스트에 닉네임과 아이디만 나오는 것이 아닌, 해당 크루의 과정 정보 + 기수 정보도 함께 나오면 좋을 것 같습니다!

---

## ✔️ 검색 조건 세부화

### 현재의 문제점
<img width="1093" alt="스크린샷 2023-04-16 오후 11 05 03" src="https://user-images.githubusercontent.com/46810837/232383799-2e737fbc-f21a-40f0-b58b-a3ee59c42043.png">

현재는 검색 시 ‘검색어를 입력하세요’라는 문구가 나오고 있어서 어떤 기준으로 검색이 가능한지 모호한 상태입니다. (실제 검색 결과는 제목 + 내용… 으로 나오는 것으로 추정되는데 현재 검색 기능이 잘 동작하고 있지 않아서 모호합니다 ㅠ)

### 개선 사항
<img width="549" alt="스크린샷 2023-04-16 오후 11 07 09" src="https://user-images.githubusercontent.com/46810837/232383855-e600a534-eb70-4bb0-95ba-495d45b3951e.png">
위 이미지는 다음 카페의 검색 조건입니다.

이런 식으로 어떤 부분에 대해서 검색을 할 수 있을지 1차적으로 지정한 다음 검색을 하게 만들거나, 이미 세션이나 미션, 태그, 멤버 기반의 세부 검색 조건이 있기 때문에 ‘검색어를 입력하세요’ 부분은 오직 제목 검색만 가능하도록 만드는 것이 더 좋을 것 같습니다.

---

## ✔️ 색의 사용이 너무 다양하다.

### 현재의 문제점

시작화면에서 “네비게이터”, “배너” 및 “인기있는 학습로그”등에 사용한 색이 너무 다양하다.
![이미지1](https://user-images.githubusercontent.com/46810837/232384114-b0588519-d30e-4303-accf-81d65b29e4ab.png)
![image](https://user-images.githubusercontent.com/46810837/232384176-240d8f97-c003-4d92-91f4-27db7f62cce0.png)

### 개선 방향

- 포인트 컬러(ex) 배너의 푸른색) 하나만 사용한다.
![image (1)](https://user-images.githubusercontent.com/46810837/232384247-027077be-0382-4b56-839e-ee47e534ead9.png)

- 흑백만 사용한다.
![image](https://user-images.githubusercontent.com/46810837/232384304-c018ba48-b227-4d25-84ad-19fa3859f255.png)

---

## ✔️ 상세 글 보기 시 다시 목록으로 돌아가기 버튼이 없다. 직접 돌아가기를 구현해야 한다

### 현재의 문제점

프롤로그 페이지 내에 뒤로가기 버튼이 없다.
![image (1)](https://user-images.githubusercontent.com/46810837/232384414-b1d75b43-b536-4486-82b5-f1828e891e6f.png)


### 개선 방향

자체 뒤로가기 버튼이 있으면 좋을거 같다.
![image](https://user-images.githubusercontent.com/46810837/232384465-e4fd50c6-45f8-4d08-b43e-b687ab3fe3ea.png)

---

## ✔️ 배너 아이콘이 블랙모드를 킬 시 보이지 않는다.

### 현재의 문제점

라이트 모드시에는 배너 아이콘이 보이지만, 다크 모드 사용자의 경우 배너 아이콘이 잘 보이지 않는다.

라이트 모드
![image (1)](https://user-images.githubusercontent.com/46810837/232384530-a55daa0a-ce7a-4070-a498-070aa37868a5.png)

다크 모드
![image (2)](https://user-images.githubusercontent.com/46810837/232384564-d6eae13b-3492-4a2e-946a-3706f0a2611e.png)

### 개선 방향

1. 라이트 모드와 다크 모드 시 서로 다른 배너 아이콘을 사용한다.
2. 동그라미 부분만 흰색으로 한다 (깃헙이 예시)

라이트 모드 시
<img src = "https://user-images.githubusercontent.com/46810837/232384635-30a534b9-5855-4e4f-8285-71e54acf638f.png" width = 200, height = 200>

다크 모드 시
<img src = "https://user-images.githubusercontent.com/46810837/232384724-c41225a0-72e8-402b-9b6d-2bd7e145592b.png" width = 200, height = 200>


---

## ✔️ 랭킹 (추가 기능)

### 현재의 문제점

“로드맵 -> 키워드 -> 답변 보러가기” 를 확인해보면 이 기능을 활용하는 크루가 많이 없다.

### 개선 방향

키워드 답변 기능 활성화를 위해 랭킹 기능을 추가.

예시)
![image (5)](https://user-images.githubusercontent.com/46810837/232384880-2146a1ac-b5ff-4ba5-8bbd-a465bc045824.png)

---

## ✔️ 적극적 활동 할 시 보상이 있으면 좋겠다. 보상 - 프로필 옆 간단한 뱃지 (추가 기능)

### 현재의 문제점

프롤로그를 사용하는 크루들만 사용하는 경향이 있음.

### 개선 방향
<img src = "https://user-images.githubusercontent.com/46810837/232385068-db58be8a-6459-4baf-885c-78b3f7f634d5.jpeg" width=300, height=400>

배민 등급 시스템 처럼 이전 달 기준으로 활동 점수(글쓰기, 댓글등)를 통해서 프로필 옆에 뱃지를 추가해주자.

---

## ✔️ 로그아웃할 시 홈 화면으로 redirect되지 않는다

### 현재의 문제점
![problem12](https://user-images.githubusercontent.com/46810837/232388471-f2494cc8-090d-4d85-bcd1-47e02f74b254.gif)


### 개선 방향
로그인 할 때와 같이 홈 화면으로 redirect해준다
![solution12-1](https://user-images.githubusercontent.com/46810837/232388661-5527ae30-966b-4cc3-83f1-c52185c71f46.gif)
![solution12-2](https://user-images.githubusercontent.com/46810837/232388947-2d5d0e6c-958a-4a01-adbb-d10eabe86831.gif)


---

## ✔️ 로드맵 화면에서 전반적인 개행이 안맞다.

### 현재의 문제점
![problem2](https://user-images.githubusercontent.com/46810837/232388225-a1765f54-4ba3-4adf-8711-d579f6bfc973.gif)


### 개선 방향 

로드맵 화면 : box 크기를 글자 크기에 맞도록 조정한다, 띄어쓰기 단위를 맞춰서 개행한다.
quiz 화면 : 1. 질문과 답변을 “ / “가 아닌 박스로 분류하던가 퀴즈 화면 내에서 카테고리를 질문과 답변을 분리한다.

---

## ✔️ Category 키워드를 클릭하지 않고 엔터를 친다면 다시 원래 화면으로 Redirect되는 문제가 있다.
이 문제를 해결하기 위해 enter를 입력할 시 redirect되지 않게 바꾼다.

### 현재의 문제점 
![problem5](https://user-images.githubusercontent.com/46810837/232388116-40693656-e816-4f1b-aaf6-3ad520ca7f62.gif)

### 개선 방향 
1.  상단에 있는 검색 입력창 처럼 enter기반으로 동작하게 만든다.
2. enter를 입력하더라도 redirect되지 않게 만든다.
![solution5](https://user-images.githubusercontent.com/46810837/232388555-1ed85427-31cb-4835-a4ac-a30d8f9b4ce9.gif)


---

## ✔️ 로그인 세션 유지가 너무 짧다

**Reply by BGuga** [reply]
정리 너무 잘하셨네요!!🤗
